support/main/default/classes/TriggerControlTest.cls

```diff
@@ -1,137 +1,243 @@
-    Boolean acctHandler1Fired,
-    Boolean acctHandler2Fired,
-    Boolean contactHandler1Fired
-  ) {
-    Id acctId = createAccount();
-    checkAccount(acctId, acctHandler1Fired, acctHandler2Fired);
-    Id contactId = createContact();
-    checkContact(contactId, contactHandler1Fired);
-  }
-
-  // NB: current version replaces all trigger controls with each call, which
-  // is plenty for the current test methods
-  private static void disable(String objectName, String handlerClassName) {
-    TriggerHandler.cacheTriggerControl(
-      new List<Trigger_Control__mdt>{
-        new Trigger_Control__mdt(
-          Object__c = objectName,
-          Class__c = handlerClassName
-        )
-      }
-    );
-  }
-
-  private static void checkContact(Id contactId, Boolean handler1Fired) {
-    Contact contact = [SELECT Handler1__c FROM Contact WHERE Id = :contactId];
-    System.assertEquals(contact.Handler1__c, handler1Fired);
-  }
-
-  private static void checkAccount(
-    Id acctId,
-    Boolean handler1Fired,
-    Boolean handler2Fired
-  ) {
-    Account acct = [
-      SELECT Handler1__c, Handler2__c
-      FROM Account
-      WHERE Id = :acctId
-    ];
-    System.assertEquals(acct.Handler1__c, handler1Fired);
-    System.assertEquals(acct.Handler2__c, handler2Fired);
-  }
-
-  private static Id createAccount() {
-    Account record = new Account(Name = 'Test');
-    insert record;
-    return record.Id;
-  }
-
-  private static Id createContact() {
-    Contact record = new Contact(LastName = 'Test');
-    insert record;
-    return record.Id;
-  }
-}
+    private static Map<String, TriggerConfigService.ObjectConfig> buildMetadata() {
+        Map<String, TriggerConfigService.ObjectConfig> mdt = new Map<String, TriggerConfigService.ObjectConfig>();
+        TriggerConfigService.ObjectConfig acct = generateObjTestData('account', true);
+        TriggerConfigService.ObjectConfig cont = generateObjTestData('contact', true);
+        acct = addChildHandlerToObj(acct, generateHandlerTestData('handler1', true));
+        acct = addChildHandlerToObj(acct, generateHandlerTestData('handler2', true));
+        cont = addChildHandlerToObj(cont, generateHandlerTestData('handler1', true));
+        mdt.put('account', acct);
+        mdt.put('contact', cont);
+        return mdt;
+    }
+
+    @isTest
+    private static void testNoControl() {
+        Map<String, TriggerConfigService.ObjectConfig> mdt = buildMetadata();
+        TriggerConfigService.configMetadata = mdt;
+        Boolean acctHandler1Fired = true;
+        Boolean acctHandler2Fired = true;
+        Boolean contactHandler1Fired = true;
+        runTest(acctHandler1Fired, acctHandler2Fired, contactHandler1Fired);
+    }
+
+    @isTest
+    private static void testContactHandler1Disable() {
+        Map<String, TriggerConfigService.ObjectConfig> mdt = buildMetadata();
+        mdt = disableHandler(mdt, 'contact', 'handler1');
+        TriggerConfigService.configMetadata = mdt;
```

Comment:
major nitpick given this is a test class

`TriggerConfigService.configMetadata = disableHandler(buildMetadata(), 'contact', 'handler1');`

alternatively add a static block since this is kind of a base before() action for this test suite
```
static {
  TriggerConfigService.configMetadata = buildMetadata();
}

someTest() { 
  disableHandler('contact', 'handler1'); // always using the same mdt ... 
}
```